### ISSUE TITLE
Broken with bash-language-server 1.5.6

### ISSUE THREAD

**Issue by raxod502** [emacs-lsp/lsp-mode#1032]
**Describe the bug**
The integration with `bash-language-server` does not work with the upgrade to version 1.5.6 of the language server.

**To Reproduce**
Install the latest version of `bash-language-server` and attempt to hover over symbols in a Bash script.

**Expected behavior**
We should get a popup window with documentation on the symbols.

**Which Language Server did you use**
`bash-ls`

**OS**
Manjaro/KDE

**Error callstack**

We have this in the `*Warnings*` buffer:

```
Warning (lsp-mode): Failed to parse the following chunk:
’’’
NEW INSTANCE TIME 1
IS CONSTRUCT CALL 1
DID NOT GOT A LOCAL
Content-Length: 96

{"jsonrpc":"2.0","method":"window/logMessage","params":{"type":4,"message":"Hovering over 3:0"}}
’’’
with message (error No Content-Length header)
```

We have this in the `*Messages*` buffer:

```
LSP :: Request textDocument/hover failed with message: Cannot read property 'rootNode' of undefined [2 times]
LSP :: Request textDocument/documentHighlight failed with message: Cannot read property 'rootNode' of undefined
LSP :: Request textDocument/hover failed with message: Cannot read property 'rootNode' of undefined
LSP :: Request textDocument/documentHighlight failed with message: Cannot read property 'rootNode' of undefined
LSP :: Request textDocument/hover failed with message: Cannot read property 'rootNode' of undefined
LSP :: Request textDocument/documentHighlight failed with message: Cannot read property 'rootNode' of undefined
LSP :: Request textDocument/hover failed with message: Cannot read property 'rootNode' of undefined [5 times]
LSP :: Request textDocument/documentHighlight failed with message: Cannot read property 'rootNode' of undefined
```

We have this in the `*lsp-log*` buffer with debugging enabled:

<details>
<p>

```
[Trace - 10:15:20 AM] Sending request 'textDocument/hover - (29)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 2,
    "character": 0
  }
}


[Trace - 10:15:20 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 2:0",
  "type": 4
}


[Trace - 10:15:20 AM] Received response 'textDocument/hover - (29)' in 7ms.
Result: null


[Trace - 10:15:21 AM] Sending request 'textDocument/hover - (30)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 2,
    "character": 0
  }
}


[Trace - 10:15:21 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 2:0",
  "type": 4
}


[Trace - 10:15:21 AM] Received response 'textDocument/hover - (30)' in 21ms.
Result: null


[Trace - 10:15:21 AM] Sending request 'textDocument/documentHighlight - (31)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 2,
    "character": 0
  }
}


[Trace - 10:15:21 AM] Received response 'textDocument/documentHighlight - (31)' in 24ms.
Result: null


[Trace - 10:15:21 AM] Sending request 'textDocument/hover - (32)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 3,
    "character": 0
  }
}


[Trace - 10:15:21 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 3:0",
  "type": 4
}


[Trace - 10:15:21 AM] Received response 'textDocument/hover - (32)' in 1ms.
Result: null


[Trace - 10:15:22 AM] Sending request 'textDocument/documentHighlight - (33)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 3,
    "character": 0
  }
}


[Trace - 10:15:22 AM] Received response 'textDocument/documentHighlight - (33)' in 0ms.
Result: null


[Trace - 10:15:23 AM] Sending request 'textDocument/hover - (34)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 0
  }
}


[Trace - 10:15:23 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 5:0",
  "type": 4
}


[Trace - 10:15:23 AM] Received response 'textDocument/hover - (34)' in 1ms.
Result: null


[Trace - 10:15:24 AM] Sending request 'textDocument/documentHighlight - (35)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 0
  }
}


[Trace - 10:15:24 AM] Received response 'textDocument/documentHighlight - (35)' in 0ms.
Result: null


[Trace - 10:15:24 AM] Sending request 'textDocument/hover - (36)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 1
  }
}


[Trace - 10:15:24 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 5:1",
  "type": 4
}


[Trace - 10:15:24 AM] Received response 'textDocument/hover - (36)' in 0ms.
Result: null


[Trace - 10:15:24 AM] Sending request 'textDocument/hover - (37)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 2
  }
}


[Trace - 10:15:24 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 5:2",
  "type": 4
}


[Trace - 10:15:24 AM] Received response 'textDocument/hover - (37)' in 0ms.
Result: null


[Trace - 10:15:25 AM] Sending request 'textDocument/hover - (38)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 3
  }
}


[Trace - 10:15:25 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 5:3",
  "type": 4
}


[Trace - 10:15:25 AM] Received response 'textDocument/hover - (38)' in 0ms.
Result: null


[Trace - 10:15:25 AM] Sending request 'textDocument/hover - (39)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 5,
    "character": 4
  }
}


[Trace - 10:15:25 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 5:4",
  "type": 4
}


[Trace - 10:15:25 AM] Received response 'textDocument/hover - (39)' in 0ms.
Result: null


[Trace - 10:15:25 AM] Sending request 'textDocument/hover - (40)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 3,
    "character": 4
  }
}


[Trace - 10:15:25 AM] Received notification 'window/logMessage'.
Params: {
  "message": "Hovering over 3:4",
  "type": 4
}


[Trace - 10:15:25 AM] Received response 'textDocument/hover - (40)' in 21ms.
Result: null


[Trace - 10:15:26 AM] Sending request 'textDocument/documentHighlight - (41)'.
Params: {
  "textDocument": {
    "uri": "file:///home/raxod502/files/control/data/emacs-repos/apheleia/scripts/docker.bash"
  },
  "position": {
    "line": 3,
    "character": 4
  }
}


[Trace - 10:15:26 AM] Received response 'textDocument/documentHighlight - (41)' in 0ms.
Result: null


```

</details>

**Comment by yyoncho** [comment]
I wasn't able to test it locally, but this should have been fixed by: c38b7eb . Please reopen if the issue is still there.

**Comment by raxod502** [comment]
Confirmed fixed, thanks! I also had to upgrade bash-language-server from 1.5.6 to 1.6.1 in order to fix the `Cannot read property 'rootNode' of undefined` errors, which required my installing from Manjaro testing -- everything works now :tada:
